### PR TITLE
[Bug] AudioFocus memory leak

### PR DESCRIPTION
**Pull request by dhiraj**
## Purpose

* Fixes leak at onFocusChange listener function var

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Open the app from Android Studio in memory profiler mode and start a call
* End the call
* Repeat several times (3-5)
* End the Activity (use hardware back button to get out of the app)
* Test mic-hold-release as per call hold feature tests (should auto-hold when another call starts, shouldn't allow resuming call when mic is not available, should auto-resume call when possible)

## What to Check
Verify that the following are valid
* Without this PR the listener function is retained and the AudioFocusManager is being held in memory

### CODE REVIEW

**Review by Mohtasim — DISMISSED** [review]
Looks good to me

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioFocusManager.kt [hunk 1]
```diff
@@ -31,34 +31,34 @@ internal abstract class AudioFocusHandler : AudioManager.OnAudioFocusChangeListe
...
     override fun releaseAudioFocus() =
-        audioManager.abandonAudioFocusRequest(audioFocusRequest26) == AudioManager.AUDIOFOCUS_REQUEST_GRANTED
+        audioManager().abandonAudioFocusRequest(audioFocusRequest26) == AudioManager.AUDIOFOCUS_REQUEST_GRANTED
 }
 
 // Legacy AudioFocus API
 @Suppress("DEPRECATION")
 internal class AudioFocusHandlerLegacy(val context: Context) : AudioFocusHandler() {
-    private val audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
+    private fun audioManager() = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
```
Comment (iaulakh) [inline azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioFocusManager.kt:51]: I think we can do same here  private fun audioManager()?

**Review by iaulakh — APPROVED** [review]

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioFocusManager.kt (hunk 1 above)
Comment (dhiraj) [reply azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioFocusManager.kt:51]: Done.

**Review by Mohtasim — APPROVED** [review]
Tested and Looks good to me.

**Pull request merged**